blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/metrics/AbstractSerDes.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.rackspacecloud.blueflood.io.serializers.metrics;
+
+import com.google.protobuf.CodedInputStream;
+import com.google.protobuf.CodedOutputStream;
+import com.rackspacecloud.blueflood.io.Constants;
+import com.rackspacecloud.blueflood.types.*;
+
+import java.io.IOException;
+
+/**
+ * This is an abstract class that most Serialization/Deserialization class
+ * extends from. It contains some utility methods.
+ */
+public abstract class AbstractSerDes {
```

Comment:
@ChandraAddala and I had a side conversation. He's ok with changing the AbstractSerDes to BaseSerDes, and cleaning up the Serializers.serializerFor() in subsequent PR.